### PR TITLE
BUG: MultiIndex.difference not keeping ea dtype

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

```
       before           after         ratio
     [b5632fb3]       [c8b6d846]
     <deprecation_notes~1>       <diff>    
-     5.67±0.08ms       4.73±0.1ms     0.83  multiindex_object.Difference.time_difference('string')
-     4.65±0.01ms      3.71±0.03ms     0.80  multiindex_object.Difference.time_difference('int')
-     4.78±0.02ms      3.75±0.01ms     0.78  multiindex_object.Difference.time_difference('ea_int')
-     6.35±0.02ms      4.85±0.02ms     0.76  multiindex_object.Difference.time_difference('datetime')
```

### CODE REVIEW

File: doc/source/whatsnew/v1.6.0.rst [hunk 1]
```diff
@@ -175,6 +175,7 @@ Missing
 
 MultiIndex
 ^^^^^^^^^^
+- Bug in :meth:`MultiIndex.difference` losing extension array dtype (:issue:`48606`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.6.0.rst:178]: Could mention the performance improvement too?

Comment (phofl) [reply doc/source/whatsnew/v1.6.0.rst:178]: Was only a bit, but could add it anyway if you prefer?

Comment (mroeschke) [reply doc/source/whatsnew/v1.6.0.rst:178]: Might as well :)

Comment (phofl) [reply doc/source/whatsnew/v1.6.0.rst:179]: Added

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**